src/process_utility.c

```diff
@@ -378,6 +393,74 @@ process_vacuum(Node *parsetree, ProcessUtilityContext context)
 
 	return true;
 }
+#else
+typedef struct VacuumCtx
+{
+	VacuumRelation *ht_vacuum_rel;
+	List	   *chunk_rels;
+} VacuumCtx;
+
+/* Adds a chunk to the list of tables to be vacuumed */
+static void
+add_chunk_to_vacuum(Hypertable *ht, Oid chunk_relid, void *arg)
+{
+	VacuumCtx  *ctx = (VacuumCtx *) arg;
+	Chunk	   *chunk = ts_chunk_get_by_relid(chunk_relid, ht->space->num_dimensions, true);
+	VacuumRelation *chunk_vacuum_rel;
+	RangeVar   *chunk_range_var = copyObject(ctx->ht_vacuum_rel->relation);
+
+	chunk_range_var->relname = NameStr(chunk->fd.table_name);
+	chunk_range_var->schemaname = NameStr(chunk->fd.schema_name);
+	chunk_vacuum_rel = makeVacuumRelation(chunk_range_var, chunk_relid, ctx->ht_vacuum_rel->va_cols);
+
+	ctx->chunk_rels = lappend(ctx->chunk_rels, chunk_vacuum_rel);
+}
+
+/* Vacuums a hypertable and all of it's chunks */
+static bool
+process_vacuum(Node *parsetree, ProcessUtilityContext context)
+{
+	VacuumStmt *stmt = (VacuumStmt *) parsetree;
+	bool		is_toplevel = (context == PROCESS_UTILITY_TOPLEVEL);
+	VacuumCtx	ctx = {
+		.ht_vacuum_rel = NULL,
+		.chunk_rels = NIL,
+	};
+	ListCell   *lc;
+	Oid			hypertable_oid;
+	Cache	   *hcache;
+	Hypertable *ht;
+	bool		affects_hypertable = false;
+
+	if (stmt->rels == NIL)
+		/* Vacuum is for all tables */
+		return false;
+
+	hcache = ts_hypertable_cache_pin();
+	foreach(lc, stmt->rels)
+	{
+		VacuumRelation *vacuum_rel = lfirst_node(VacuumRelation, lc);
+
+		hypertable_oid = ts_hypertable_relid(vacuum_rel->relation);
```

Comment:
a bit weird to use `ts_hypertable_relid` before using `ts_hypertable_cache_get_entry` I would use `ts_hypertable_cache_get_entry` passing it the relid from `RangeVarGetRelid(vacuum_rel->relation, NoLock, true)`

Reply:
this is something that was there before I just used the same convention as the previous function. I think a small new commit to fix this would be warranted, but we talked and you said that it should be separate from the PG11 changes.